src/panels/config/automation/action/types/ha-automation-action-wait_for_trigger.ts

```diff
@@ -52,6 +55,17 @@ export class HaWaitForTriggerAction
     `;
   }
 
+  private _timeoutChanged(ev: CustomEvent): void {
+    ev.stopPropagation();
+    const value = ev.detail.value;
+    if (!value) {
+      return;
+    }
+    fireEvent(this, "value-changed", {
+      value: { ...this.action, timeout: value },
+    });
+  }
+
   private _continueChanged(ev) {
     fireEvent(this, "value-changed", {
       value: { ...this.action, continue_on_timeout: ev.target.checked },
```

Comment:
You can use `CustomEveent<DetailType>` to better type this.

Reply:
That would be changing existing code. As this code isn't changed or touched in this PR.